### ISSUE TITLE
[I3OmDb] Not compiling for icerec/V05-00-02 (Trac #1853)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1845]
Does not compile for me when using clang however gcc works. Forward declaration wrongly executed.

```text
[ 88%] Building CXX object STTools/CMakeFiles/STTools-pybindings.dir/private/pybindings/algorithms/seededRT/I3SeededRTConfigurationService.cxx.o
In file included from /home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/private/I3OmDb/I3OmDbFactory.cxx:2:
/home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/public/I3Db/I3OmDb/I3OmDb.h:1280:8: error: 'Om_s' redeclared with 'private' access
struct  Om_s 
        ^
/home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/public/I3Db/I3OmDb/I3OmDb.h:977:8: note: previously declared 'public' here
struct Om_s;
       ^
1 error generated.
make[2]: *** [I3Db/CMakeFiles/I3Db.dir/private/I3OmDb/I3OmDbFactory.cxx.o] Error 1
make[2]: Target `I3Db/CMakeFiles/I3Db.dir/build' not remade because of errors.
make[1]: *** [I3Db/CMakeFiles/I3Db.dir/all] Error 2
```

Migrated from https://code.icecube.wisc.edu/ticket/1853
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:24", 
    "description": "Does not compile for me when using clang however gcc works. Forward declaration wrongly executed.\n\n{{{\n[ 88%] Building CXX object STTools/CMakeFiles/STTools-pybindings.dir/private/pybindings/algorithms/seededRT/I3SeededRTConfigurationService.cxx.o\nIn file included from /home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/private/I3OmDb/I3OmDbFactory.cxx:2:\n/home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/public/I3Db/I3OmDb/I3OmDb.h:1280:8: error: 'Om_s' redeclared with 'private' access\nstruct  Om_s \n        ^\n/home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/public/I3Db/I3OmDb/I3OmDb.h:977:8: note: previously declared 'public' here\nstruct Om_s;\n       ^\n1 error generated.\nmake[2]: *** [I3Db/CMakeFiles/I3Db.dir/private/I3OmDb/I3OmDbFactory.cxx.o] Error 1\nmake[2]: Target `I3Db/CMakeFiles/I3Db.dir/build' not remade because of errors.\nmake[1]: *** [I3Db/CMakeFiles/I3Db.dir/all] Error 2\n}}}\n", 
    "reporter": "maryon", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067204154158", 
    "component": "combo reconstruction", 
    "summary": "[I3OmDb] Not compiling for icerec/V05-00-02", 
    "priority": "blocker", 
    "keywords": "I3OmDb icerec", 
    "time": "2016-09-09T11:50:36", 
    "milestone": "", 
    "owner": "joertlin", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160909T16:01:56: 
* nega commented:

> Development on I3Db has ceased. If you don't need it, remove it from your source directory and re-cmake and re-make. If you do need it, and can't use `gcc`, please re-open the ticket.


* nega changed resolution from "" to "wontfix"
* nega changed description from:

> Does not compile for me when using clang however gcc works. Forward declaration wrongly executed.
> 
> [ 88%] Building CXX object STTools/CMakeFiles/STTools-pybindings.dir/private/pybindings/algorithms/seededRT/I3SeededRTConfigurationService.cxx.o
> In file included from /home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/private/I3OmDb/I3OmDbFactory.cxx:2:
> /home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/public/I3Db/I3OmDb/I3OmDb.h:1280:8: error: 'Om_s' redeclared with 'private' access
> struct  Om_s 
>         ^
> /home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/public/I3Db/I3OmDb/I3OmDb.h:977:8: note: previously declared 'public' here
> struct Om_s;
>        ^
> 1 error generated.
> make[2]: *** [I3Db/CMakeFiles/I3Db.dir/private/I3OmDb/I3OmDbFactory.cxx.o] Error 1
> make[2]: Target `I3Db/CMakeFiles/I3Db.dir/build' not remade because of errors.
> make[1]: *** [I3Db/CMakeFiles/I3Db.dir/all] Error 2
> 

to:

> Does not compile for me when using clang however gcc works. Forward declaration wrongly executed.
> 
> {{{
> [ 88%] Building CXX object STTools/CMakeFiles/STTools-pybindings.dir/private/pybindings/algorithms/seededRT/I3SeededRTConfigurationService.cxx.o
> In file included from /home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/private/I3OmDb/I3OmDbFactory.cxx:2:
> /home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/public/I3Db/I3OmDb/I3OmDb.h:1280:8: error: 'Om_s' redeclared with 'private' access
> struct  Om_s 
>         ^
> /home/maryon/software/meta-projects/icerec/V05-00-02/src/I3Db/public/I3Db/I3OmDb/I3OmDb.h:977:8: note: previously declared 'public' here
> struct Om_s;
>        ^
> 1 error generated.
> make[2]: *** [I3Db/CMakeFiles/I3Db.dir/private/I3OmDb/I3OmDbFactory.cxx.o] Error 1
> make[2]: Target `I3Db/CMakeFiles/I3Db.dir/build' not remade because of errors.
> make[1]: *** [I3Db/CMakeFiles/I3Db.dir/all] Error 2
> }}}
> 


* nega changed status from "new" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20160909T16:10:20: 
* olivas commented:

> I3Db needed to be included in the last icerec release.  We can make this build with clang.


* olivas changed status from "closed" to "reopened"
* olivas changed resolution from "wontfix" to ""

**Comment by icecube-trac** [comment]
Trac update at 20160909T16:18:41: 
* joertlin changed owner from "olivas" to "joertlin"
* joertlin changed status from "reopened" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20160909T21:07:16: 
* joertlin changed resolution from "" to "fixed"
* joertlin commented:

> In changeset 149921:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="149921"
> My last commit that added IceACT contained an error: I started using the private struct I3OmDb::Om_s outside of I3OmDb. Since I also added a forward declaration for this struct in a public environment, gcc compiled w/o an error. However, clang complains. Therefore, I made I3OmDb::Om_s public. Closes #1845
> ```


* joertlin changed status from "accepted" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:24: 
* olivas commented:

> Milestone October 2016 deleted


* olivas changed milestone from "October 2016" to ""
